mantidproject/mantid
# Expose save_image and replace_workspace to the python api of Instrument Viewer

**Description of work.**
The title says it all.



**To test:**

Run the script below by substituting the input and output directories and run numbers for raw data of your choice. 
Any data should do as long as it is loaded to a MatrixWorkspace with instrument geometry attached.
Finally, check in your specified output directory that the .png images are created and look as nice.

```
# import mantid algorithms, numpy and matplotlib
from mantid.simpleapi import *
import matplotlib.pyplot as plt
from mantidqt.widgets.instrumentview.api import get_instrumentview
import numpy as np
from time import sleep

def show_instrument_view(runstart, runend, indir, outdir):
    config.appendDataSearchDir(indir)
    Load(Filename=str(runstart),OutputWorkspace=str(runstart))
    iv = get_instrumentview(runstart)
    iv.show_view()
    run = runstart + 1
    while run < runend:
        Load(Filename=str(run), OutputWorkspace=str(run))
        iv.replace_workspace(str(run))
        run += 1
        iv.save_image(outdir+str(run)+'.png')
    
input_directory = '/Volumes/illdata/212/d11/internalUse/rawdata'
output_directory = '/Users/vardanyan/Desktop/thumbnails/'
runstart = 8539
runend = 8560
   
show_instrument_view(runstart, runend, input_directory, output_directory)
```

*There is no associated issue.*





---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
